scripts/ci/check_compliance.py

```diff
@@ -1446,44 +1445,36 @@ class Identity(ComplianceTest):
 
     def run(self):
         for shaidx in get_shas(COMMIT_RANGE):
-            commit = git("log", "--decorate=short", "--no-use-mailmap", "-n 1", shaidx)
-            signed = []
-            author = ""
-            sha = ""
-            parsed_addr = None
-            for line in commit.split("\n"):
-                match = re.search(r"^commit\s([^\s]*)", line)
-                if match:
-                    sha = match.group(1)
-                match = re.search(r"^Author:\s(.*)", line)
-                if match:
-                    author = match.group(1)
-                    parsed_addr = parseaddr(author)
-                match = re.search(r"signed-off-by:\s(.*)", line, re.IGNORECASE)
-                if match:
-                    signed.append(match.group(1))
-
-            error1 = f"{sha}: author email ({author}) needs to match one of " \
-                     f"the signed-off-by entries."
-            error2 = f"{sha}: author email ({author}) does not follow the " \
-                     f"syntax: First Last <email>."
-            error3 = f"{sha}: author email ({author}) must be a real email " \
-                     f"and cannot end in @users.noreply.github.com"
-            failure = None
-            if author not in signed:
-                failure = error1
-
-            if not parsed_addr or len(parsed_addr[0].split(" ")) < 2:
-                if not failure:
-
-                    failure = error2
-                else:
-                    failure = failure + "\n" + error2
-            elif parsed_addr[1].endswith("@users.noreply.github.com"):
-                failure = error3
-
-            if failure:
-                self.failure(failure)
+            lines = git('show', '-s', '--format=%an%n%ae%n%b',
+                        shaidx).split('\n')
+            auth_name, auth_email, *body = lines
+            body = '\n'.join(body)
```

Comment:
Can be done in a single call.

```suggestion
            auth_name, auth_email, body = git('show', '-s', '--format=%an%n%ae%n%b', 
                                              shaidx).split('\n', 2)
```

Reply:
Nice catch, was unaware there was an optional maxsplit argument to split!